client/ui/components/addon-groups/SyncDiff.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { PropTypes as pt } from 'react';
+
+import RaisedButton from 'material-ui/RaisedButton';
+import { Toolbar, ToolbarGroup } from 'material-ui/Toolbar';
+
+import LoadingIndicator from '../indicators/LoadingIndicator';
+import containSyncDiff from '../../containers/SyncDiffContainer';
+import QueryAddonGroups from '../data/QueryAddonGroups';
+import goTo from '../../utils/goTo';
+
+
+const style = {
+  toolbar: {
+    justifyContent: 'flex-end',
+  },
+};
+
+
+class SyncDiff extends React.Component {
+  static propTypes = {
+    addonGroups: pt.array,
+    count: pt.number,
+    type: pt.string,
+  };
+
+  mapAddons(addonGroup) {
```

Comment:
This smells more like a new component than a method to me, one that could take in the diff as a prop instead of the type and the group.